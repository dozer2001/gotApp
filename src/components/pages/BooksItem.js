import React, {Component} from 'react';
import ItemDetails,{Field} from '../itemDetails/index';
import gotService from '../services/gotServices';

export default class BooksItem extends Component{
    gotService = new gotService();



    render(){

        return(
            <ItemDetails id = {this.props.bookId}
                         getData={this.gotService.getBook}>
                <Field field='name' label='Name'/>
                <Field field='numberOfPages' label='NumberOfPages'/>
                <Field field='publiser' label='Publiser'/>
                <Field field='released' label='Released'/>
            </ItemDetails>
        )

    }
}