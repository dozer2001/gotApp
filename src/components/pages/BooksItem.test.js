import React from 'react';
import {shallow} from 'enzyme';
import BooksItem from './BooksItem';


describe('Testing <HousesPage/>', () => {
    const item = shallow(<BooksItem/>);
    it('houses have rendered correctly', () => {
        expect(item).toMatchSnapshot();
    });
});