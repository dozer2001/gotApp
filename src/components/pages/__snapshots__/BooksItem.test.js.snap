// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HousesPage/> houses have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BooksItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Field
          field="name"
          label="Name"
        />,
        <Field
          field="numberOfPages"
          label="NumberOfPages"
        />,
        <Field
          field="publiser"
          label="Publiser"
        />,
        <Field
          field="released"
          label="Released"
        />,
      ],
      "getData": [Function],
      "id": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "field": "name",
          "label": "Name",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "field": "numberOfPages",
          "label": "NumberOfPages",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "field": "publiser",
          "label": "Publiser",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "field": "released",
          "label": "Released",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Field
            field="name"
            label="Name"
          />,
          <Field
            field="numberOfPages"
            label="NumberOfPages"
          />,
          <Field
            field="publiser"
            label="Publiser"
          />,
          <Field
            field="released"
            label="Released"
          />,
        ],
        "getData": [Function],
        "id": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "field": "name",
            "label": "Name",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "field": "numberOfPages",
            "label": "NumberOfPages",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "field": "publiser",
            "label": "Publiser",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "field": "released",
            "label": "Released",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
